browser/src/shared/code-hosts/shared/util/fileInfo.ts

```diff
@@ -4,59 +4,90 @@ import { catchError, map } from 'rxjs/operators'
-        retryWhenCloneInProgressError()
-    )
-
-    const requests = [resolvingHeadRev]
-
-    // If theres a base, resolve it as well.
-    if (baseCommitID) {
-        const resolvingBaseRev = resolveRev({ repoName, rev: baseCommitID, requestGraphQL }).pipe(
-            retryWhenCloneInProgressError()
-        )
-        requests.push(resolvingBaseRev)
-    }
-
-    return zip(...requests).pipe(map(() => ({ repoName, commitID, baseCommitID, ...rest })))
-}
+/**
+ * Use `rawRepoName` for the value of `repoName`, as a fallback if `repoName` was not available.
+ * */
+const useRawRepoNameAsFallback = (fileInfo: FileInfo): FileInfoWithRepoName => ({
+    ...fileInfo,
+    repoName: fileInfo.rawRepoName,
+})
 
 /**
  * Resolve a `FileInfo`'s raw repo names to their Sourcegraph
  * repo names as affected by `repositoryPathPattern`.
  */
-export const resolveRepoNames = (
-    { rawRepoName, baseRawRepoName, ...rest }: FileInfo,
+const resolveRepoNameForFileInfo = (
+    fileInfo: FileInfo,
     requestGraphQL: PlatformContext['requestGraphQL']
-): Observable<FileInfoWithRepoNames> => {
-    const resolvingHeadRepoName = resolveRepo({ rawRepoName, requestGraphQL }).pipe(retryWhenCloneInProgressError())
-    const resolvingBaseRepoName = baseRawRepoName
-        ? resolveRepo({ rawRepoName: baseRawRepoName, requestGraphQL }).pipe(retryWhenCloneInProgressError())
-        : of(undefined)
-
-    return zip(resolvingHeadRepoName, resolvingBaseRepoName).pipe(
-        map(([repoName, baseRepoName]) => ({ repoName, baseRepoName, rawRepoName, baseRawRepoName, ...rest })),
-
-        // ERPRIVATEREPOPUBLICSOURCEGRAPHCOM likely means that the user is viewing private code
-        // without having pointed his browser extension to a self-hosted Sourcegraph instance that
-        // has access to that code. In that case, it's impossible to resolve the repo names,
-        // so we keep the repo names inferred from the code host's DOM.
+): Observable<FileInfoWithRepoName> =>
+    resolveRepo({ rawRepoName: fileInfo.rawRepoName, requestGraphQL }).pipe(
+        map(repoName => ({ ...fileInfo, repoName })),
         catchError(err => {
+            // ERPRIVATEREPOPUBLICSOURCEGRAPHCOM likely means that the user is viewing private code
+            // without having pointed his browser extension to a self-hosted Sourcegraph instance that
+            // has access to that code. In that case, it's impossible to resolve the repo names,
+            // so we keep the repo names inferred from the code host's DOM.
             if (isPrivateRepoPublicSourcegraphComErrorLike(err)) {
-                return [{ rawRepoName, baseRawRepoName, repoName: rawRepoName, baseRepoName: baseRawRepoName, ...rest }]
+                return of(useRawRepoNameAsFallback(fileInfo))
             }
             throw err
         })
     )
+
+export const ensureRev = <T extends FileInfo>(fileInfo: T): T & { rev: string } => ({
```

Comment:
Suggestion for a more precise name
```suggestion
export const defaultRevToCommitID = <T extends FileInfo>(fileInfo: T): T & { rev: string } => ({
```